core/src/main/java/com/graphhopper/routing/subnetwork/PrepareRoutingSubnetworks.java

```diff
@@ -22,24 +22,36 @@
 import com.carrotsearch.hppc.cursors.IntCursor;
 import com.graphhopper.routing.ev.BooleanEncodedValue;
-import com.graphhopper.routing.util.EdgeFilter;
 import com.graphhopper.routing.weighting.Weighting;
 import com.graphhopper.storage.GraphHopperStorage;
-import com.graphhopper.util.*;
+import com.graphhopper.util.EdgeIteratorState;
+import com.graphhopper.util.GHUtility;
+import com.graphhopper.util.Helper;
+import com.graphhopper.util.StopWatch;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.util.ArrayList;
 import java.util.List;
 
 /**
- * Removes nodes/edges which are not part of the 'main' network(s). I.e. mostly nodes with no edges at all but
- * also small subnetworks which could be bugs in OSM data or 'islands' or indicate otherwise disconnected areas
- * e.g. via barriers or one way problems - see #86. Subnetworks are removed by disabling access to the corresponding
- * edges for a given access encoded value. It is important to search for strongly connected components here (i.e.
- * consider that the graph is directed). For example, small areas like parking lots are sometimes connected to the whole
- * network through a single one-way road (a mapping error) and have to be removed because otherwise the routing fails
- * when starting from such a parking lot.
+ * Detects and marks 'subnetworks' with a dedicated subnetwork encoded value. Subnetworks are parts of the road network
+ * that are not connected to the rest of the network and that are below a certain size. These can be isolated nodes with
+ * no edges at all, but also small subnetworks which could be bugs in OSM data or 'islands' that are separated from
+ * the rest of the network because of some missing link, barrier or some closed road for example.
+ * <p>
+ * Sometimes there are also subnetworks that can be reached from the main network but not the other way around (or the
+ * opposite). For example this can be parking lots that can only be accessed by a single one-way road (a mapping error).
+ * These are called 'one-way subnetworks' and are marked using the same subnetwork encoded value, see #86. To find such
+ * one-way subnetworks it is important to search for strongly connected components on the directed graph and not do a
+ * simple connectivity check for one direction.
+ * <p>
+ * Note that it depends on the weighting whether or not edges belong to a subnetwork or not. For example if a weighting
+ * 'closes' a bridge to an island the island might become a subnetwork, but if the bridge was open it would belong to
+ * the main network. There can even be subnetworks that are due to turn restrictions.
+ * <p>
+ * We always run an edge-based connected component search, because this way we retrieve the edges (not the nodes) that
+ * belong to each component and can include turn restrictions as well. Node-based component search is faster, but since
+ * the subnetwork search goes relatively fast anyway using it has no real benefit.
```

Comment:
I updated the docs and tried to get rid of everything that mentioned the 'removal' of edges or nodes, because we do not (or do no longer) remove anything. All this class now does is it sets the subnetwork encoded value.

Reply:
Thanks!